lib/client/TimeScaler.ts

```diff
@@ -22,6 +22,9 @@ export class TimeScaler {
     this.gridColScale = scaleTime().domain(domain).range([0, numCols]);
   }
 
+  getPercentileInverse(num: number) {
+    return this.percentageScale.invert(num)
+  }
```

Comment:
could be useful elsewhere, but mostly to help with places where we need to determine if the scale has changed. To check, you can always pass the result of `dayjs(globalTimeScaler.getPercentileInverse(0.5)).format('YYYY-MM-DD')` into your useEffect dependencies list.